src/mantle/framework/class-bootloader.php

```diff
@@ -0,0 +1,275 @@
+		);
+
+		$this->app->singleton_if(
+			Contracts\Exceptions\Handler::class,
+			\Mantle\Framework\Exceptions\Handler::class,
+		);
+
+		/**
+		 * Fired after the application is booted.
+		 *
+		 * @param \Mantle\Contracts\Application $app Application instance.
+		 */
+		$this->app['events']->dispatch( 'mantle_bootloader_booted', $this->app );
+	}
+
+	/**
+	 * Retrieve the application instance.
+	 *
+	 * @return Contracts\Application|null
+	 */
+	public function get_application(): ?Contracts\Application {
+		return $this->app;
+	}
+
+	/**
+	 * Get the calculated base path for the application.
+	 *
+	 * @todo Calculate a better default path from the plugin file.
+	 *
+	 * @return string|null
+	 */
+	public function get_base_path(): ?string {
+		if ( ! empty( $this->base_path ) ) {
+			return $this->base_path;
+		}
+
+		return match ( true ) {
+			! empty( $_ENV['MANTLE_BASE_PATH'] ) => $_ENV['MANTLE_BASE_PATH'],
+			defined( 'MANTLE_BASE_PATH' ) => constant( 'MANTLE_BASE_PATH' ),
+			default => dirname( __DIR__, 3 ),
+		};
+	}
+
+	/**
+	 * Set the base path for the application.
+	 *
+	 * @param string|null $base_path Base path for the application.
+	 * @return static
+	 */
+	public function set_base_path( ?string $base_path = null ): static {
+		$this->base_path = $base_path;
+
+		return $this;
+	}
+
+	/**
+	 * Boot the application in the console context.
+	 *
+	 * @return void
+	 */
+	protected function boot_console(): void {
+		$kernel = $this->app->make( Contracts\Console\Kernel::class );
+
+		$status    = $kernel->handle(
+			$input = new \Symfony\Component\Console\Input\ArgvInput(),
+			new \Symfony\Component\Console\Output\ConsoleOutput(),
+		);
+
+		$kernel->terminate( $input, $status );
+
+		exit( (int) $status );
+	}
+
+	/**
+	 * Boot the application in the WP-CLI context.
+	 *
+	 * @return void
+	 */
+	protected function boot_console_wp_cli(): void {
+		\WP_CLI::add_command(
+			/**
+			 * Command prefix for Mantle WP-CLI commands.
+			 *
+			 * @param string $prefix The command prefix.
+			 * @param \Mantle\Contracts\Application $app The application instance.
+			 */
+			(string) apply_filters( 'mantle_console_command_prefix', Command::PREFIX, $this->app ),
+			function () {
+				$kernel = $this->app->make( Contracts\Console\Kernel::class );
+
+				$status    = $kernel->handle(
+					$input = new \Symfony\Component\Console\Input\ArgvInput(
+						collect( (array) ( $_SERVER['argv'] ?? [] ) ) // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+							// Remove the `wp` prefix from argv and any invalid arguments (such as --url).
+							->filter( fn ( $value, $index ) => 0 !== $index && ! Str::starts_with( $value, '--url=' ) )
+							->all()
+					),
+					new \Symfony\Component\Console\Output\ConsoleOutput(),
+				);
+
+				$kernel->terminate( $input, $status );
+
+				exit( (int) $status );
+			},
+			[
+				'shortdesc' => __( 'Mantle Framework Command Line Interface', 'mantle' ),
+			]
+		);
+	}
+
+	/**
```

Comment:
Do we anticipate other contexts where we want to boot like web sockets or using a wp-now type setup or something else? Do we want to make this more extensible?

Reply:
`wp-now` would fall into a HTTP request but sockets would be something to think about